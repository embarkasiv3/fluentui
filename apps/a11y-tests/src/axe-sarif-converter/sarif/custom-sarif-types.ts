// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
export const enum SarifLogVersion {
  v2 = '2.0.0'
}

export namespace Result {
  export const enum level {
    notApplicable = 'notApplicable',
    pass = 'pass',
    note = 'note',
    warning = 'warning',
    error = 'error',
    open = 'open'
  }
}

/**
 * Encapsulates a message intended to be read by the end user.
 */
// tslint:disable-next-line:interface-name
export interface Message {
  /**
   * A plain text message string.
   */
  text?: string;

  /**
   * The resource id for a plain text message string.
   */
  messageId?: string;

  /**
   * A rich text message string.
   */
  richText?: string;

  /**
   * The resource id for a rich text message string.
   */
  richMessageId?: string;

  /**
   * An array of strings to substitute into the message string.
   */
  arguments?: string[];
}
