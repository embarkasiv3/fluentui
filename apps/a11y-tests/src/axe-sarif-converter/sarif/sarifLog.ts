// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { StaticAnalysisResultsFormatSarifVersion200JsonSchema } from './sarifv2';
export interface SarifLog extends StaticAnalysisResultsFormatSarifVersion200JsonSchema {}
