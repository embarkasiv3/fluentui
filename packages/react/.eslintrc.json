{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
