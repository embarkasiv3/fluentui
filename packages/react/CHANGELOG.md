# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
