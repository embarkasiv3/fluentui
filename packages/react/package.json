{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
