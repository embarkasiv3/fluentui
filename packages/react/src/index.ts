import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
