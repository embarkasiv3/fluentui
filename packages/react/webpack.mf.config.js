const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

const myConfig = {
  output: {},
  entry: {},
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    contentBase: 'dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
