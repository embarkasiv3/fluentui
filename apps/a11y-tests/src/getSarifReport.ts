import { AxePuppeteer } from 'axe-puppeteer';
import * as puppeteer from 'puppeteer';
import { convertAxeToSarif } from 'axe-sarif-converter';
import { SarifLog } from 'axe-sarif-converter/dist/sarif/sarif-log'; // TODO - merge with prev line when SarifLog is exported from index

const TEST_URL_ROOT = process.env.BUILD_SOURCEBRANCH
  ? `http://fabricweb.z5.web.core.windows.net/pr-deploy-site/${process.env.BUILD_SOURCEBRANCH}/fabric-website-resources/dist/`
  : 'http://localhost:4322';

export async function getSarifReport(subUrl: string): Promise<SarifLog> {
  const browser = await puppeteer.launch();

  const page = await browser.newPage();
  await page.setBypassCSP(true);

  try {
    await page.goto(`${TEST_URL_ROOT}${subUrl}`, { waitUntil: 'load', timeout: 60000 });
  } catch (e) {
    browser.close();
    throw e;
  }

  const axeReport = await new AxePuppeteer(page).include(['.ExampleCard-example']).analyze();
  const sarifReport = convertAxeToSarif(axeReport);

  await page.close();
  await browser.close();

  return sarifReport;
}
