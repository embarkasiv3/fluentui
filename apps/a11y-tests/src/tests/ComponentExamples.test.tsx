import * as React from 'react';
import * as fs from 'fs';
import * as path from 'path';
import * as glob from 'glob';
import { getSarifReport } from '../getSarifReport';
import { SarifLog } from 'axe-sarif-converter/dist/sarif/sarif-log';
import { Result } from 'axe-sarif-converter/dist/sarif/sarif-2.0.0';

// Keep only errors to reduce snapshot size
function dehydrateSarifReport(report: SarifLog): Result[] {
  return report.runs[0]!.results!.filter(item => item.level === 'error');
}

/* tslint:disable-next-line:no-any */
async function testComponent(component: { name: string; pageName: string; elem: React.ReactElement<any> }) {
  it(`checks accessibility of ${component.name} (${component.pageName})`, async () => {
    const sarifReport: SarifLog = await getSarifReport(component.elem);

    // Save the report into `dist/reports` folder
    fs.writeFileSync(path.resolve(__dirname, `../../dist/reports/${component.pageName}.sarif`), JSON.stringify(sarifReport), {
      encoding: 'utf8'
    });

    // Match the 'errors' section with snapshot
    expect(dehydrateSarifReport(sarifReport)).toMatchSnapshot();
  });
}

function getControlAndPageName(exampleFilePath: string): [string, string] {
  const match = exampleFilePath.match(/components\/(.+)\/examples\/(.+)\.js/)!;
  return [match[1], match[2]];
}

// List of controls we expose to a11y tests
const enabledControls: string[] = ['Button', 'TextField'];
const files: string[] = [];

enabledControls.forEach((control: string) => {
  const exampleFiles: string[] = glob.sync(
    path.resolve(process.cwd(), `node_modules/office-ui-fabric-react/lib-commonjs/components/${control}/examples/*Example*.js`)
  );
  files.push(...exampleFiles);
});

files.forEach((componentFile: string) => {
  const componentModule = require(componentFile);
  const [controlName, pageName] = getControlAndPageName(componentFile);
  Object.keys(componentModule)
    .filter(key => typeof componentModule[key] === 'function')
    .forEach(key => {
      const ComponentUnderTest: React.ComponentClass = componentModule[key];
      testComponent({
        name: controlName,
        pageName: pageName,
        elem: <ComponentUnderTest />
      });
    });
});
