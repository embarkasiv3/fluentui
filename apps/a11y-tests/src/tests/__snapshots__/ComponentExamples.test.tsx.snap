// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks accessibility of Button (Button.Action.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Anchor.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Command.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.CommandBar.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Compound.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.ContextualMenu.Example) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<button type=\\"button\\" data-automation-id=\\"test\\" class=\\"ms-Button ms-Button--default root-1\\" data-is-focusable=\\"true\\" aria-expanded=\\"false\\" aria-owns=\\"id__3-menu\\" aria-haspopup=\\"true\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "button[aria-owns=\\"id__3-menu\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- Invalid ARIA attribute value: aria-owns=\\"id__3-menu\\"",
      "text": "Fix all of the following: Invalid ARIA attribute value: aria-owns=\\"id__3-menu\\".",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "button[aria-owns=\\"id__3-menu\\"]",
      "ruleId": "aria-valid-attr-value",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-valid-attr-value",
  },
]
`;

exports[`checks accessibility of Button (Button.Default.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Icon.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Primary.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.ScreenReader.Example) 1`] = `Array []`;

exports[`checks accessibility of Button (Button.Split.Example) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-10\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(1) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(1) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-10\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(2) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(2) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-10\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "div:nth-child(3) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "div:nth-child(3) > .css-10[data-is-focusable=\\"true\\"][role=\\"button\\"]",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" aria-disabled=\\"true\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-35\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".css-35",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".css-35",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
]
`;

exports[`checks accessibility of Button (Button.Split.Example) 2`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<div data-automation-id=\\"test\\" aria-roledescription=\\"split button\\" data-is-focusable=\\"true\\" role=\\"button\\" aria-haspopup=\\"true\\" aria-expanded=\\"false\\" class=\\"css-10\\" tabindex=\\"0\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".css-10",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription",
      "text": "Fix all of the following: ARIA attribute is not widely supported in screen readers and assistive technologies:  aria-roledescription.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".css-10",
      "ruleId": "aria-allowed-attr",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "aria-allowed-attr",
  },
]
`;

exports[`checks accessibility of Button (Button.Toggle.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.Basic.Example) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField8\\" required=\\"\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField8",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Form element does not have an implicit (wrapped) &lt;label>
- Form element does not have an explicit &lt;label>
- Element has no title attribute or the title attribute is empty",
      "text": "Fix any of the following: aria-label attribute does not exist or is empty. aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty. Form element does not have an implicit (wrapped) <label>. Form element does not have an explicit <label>. Element has no title attribute or the title attribute is empty.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField8",
      "ruleId": "label",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label",
  },
]
`;

exports[`checks accessibility of TextField (TextField.Borderless.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.Controlled.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.CustomRender.Example) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField2\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-describedby=\\"TextFieldDescription3\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField2",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix all of the following:
- Only title used to generate label for form element",
      "text": "Fix all of the following: Only title used to generate label for form element.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField2",
      "ruleId": "label-title-only",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label-title-only",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<input type=\\"text\\" id=\\"TextField2\\" value=\\"\\" class=\\"ms-TextField-field field-6\\" aria-describedby=\\"TextFieldDescription3\\" aria-invalid=\\"false\\">",
            },
          },
        ],
        "fullyQualifiedLogicalName": "#TextField2",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- aria-label attribute does not exist or is empty
- aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Form element does not have an implicit (wrapped) &lt;label>
- Form element does not have an explicit &lt;label>
- Element has no title attribute or the title attribute is empty",
      "text": "Fix any of the following: aria-label attribute does not exist or is empty. aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty. Form element does not have an implicit (wrapped) <label>. Form element does not have an explicit <label>. Element has no title attribute or the title attribute is empty.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": "#TextField2",
      "ruleId": "label",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "label",
  },
]
`;

exports[`checks accessibility of TextField (TextField.ErrorMessage.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.Masked.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.Multiline.Example) 1`] = `Array []`;

exports[`checks accessibility of TextField (TextField.PrefixAndSuffix.Example) 1`] = `
Array [
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".prefix-16 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".prefix-16 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">.com</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(1) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">https://</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .prefix-10.ms-TextField-prefix > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
  Object {
    "level": "error",
    "locations": Array [
      Object {
        "annotations": Array [
          Object {
            "snippet": Object {
              "text": "<span style=\\"padding-bottom:1px\\">.com</span>",
            },
          },
        ],
        "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
        "physicalLocation": Object {
          "fileLocation": Object {
            "uri": "about:blank",
          },
        },
      },
    ],
    "message": Object {
      "richText": "Fix any of the following:
- Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1",
      "text": "Fix any of the following: Element has insufficient color contrast of 2.43 (foreground color: #a6a6a6, background color: #ffffff, font size: 10.5pt, font weight: normal). Expected contrast ratio of 4.5:1.",
    },
    "partialFingerprints": Object {
      "fullyQualifiedLogicalName": ".ms-TextField.root-3:nth-child(2) > .ms-TextField-wrapper.wrapper-4 > .fieldGroup-5.ms-TextField-fieldGroup > .ms-TextField-suffix.suffix-11 > span",
      "ruleId": "color-contrast",
    },
    "properties": Object {
      "tags": Array [
        "Accessibility",
      ],
    },
    "ruleId": "color-contrast",
  },
]
`;

exports[`checks accessibility of TextField (TextField.Styled.Example) 1`] = `Array []`;
